docs/service_x509_credentials_aws.md

```diff
@@ -0,0 +1,346 @@
+and `<ec2-instance-role>` values with their corresponding values for
+your environment:
+
+```
+ {
+   "Effect": "Allow",
+   "Principal": {
+     "AWS": "arn:aws:iam::<account-id>:role/<ec2-instance-role>"
+   },
+   "Action": "sts:AssumeRole"
+ }
+```
+
+Once correctly updated, your EC2 instance role must appear in the `Trusted entities` table:
+
+![IAM Role Setup_3](images/iam-role-setup-3.png)
+
+### Setup for Services with Region Specific Instance Profile Roles
+
+The IAM Role setup is identical when you have the same service
+being deployed in multiple regions with role names that
+are specific to each region. Since IAM roles are global, you only need
+to create a single Athenz IAM Role in the `<domain>.<service>` format.
+Then, when you're setting up your trust relationship, you'll authorize
+all your region specific profile roles to have the capability to
+assume the Athenz `<domain>.<service>` role.
+
+## Installing SIA
+
+The AWS SIA source is part of the Athenz project and can be found in:
+```
+provider/aws/sia-ec2
+``` 
+Follow the readme for instructions on how to install it.
+
+## SIA Configuration Setup
+
+When building your image, you can include the following configuration
+file called `sia_config` in the `/etc/sia` directory. It must include
+the following required fields:
+
+```
+{
+  "version": "1.0.0",
+  "service": "<service name>",
+  "accounts": [
+    {
+      "domain": "<domain name>",
+      "account": "<aws account id associated with domain name>"
+    }
+  ]
+}
+```
+
+The domain/account/service values here must match to the IAM Role
+created earlier. For example, if the service identity name is `api`
+in domain `sports` whose corresponding aws account id is `123456789`,
+then the `sia_config` file will be as follows:
+
+```
+{
+  "version": "1.0.0",
+  "service": "api",
+  "accounts": [
+    {
+      "domain": "sports",
+      "account": "123456789"
+    }
+  ]
+}
+```
+
+If you require bastion ssh access to your instances, then the profile
+role must be called `<domain>.<service>-service` and you don't need to
+specify this configuration file unless you want to change the ownership
+of the generated private key and certificate files - see
+[Private Key File Ownership](#private-key-file-ownership) section for more details.
+
+### Private Key File Ownership
+
+By default SIA agent runs as root and the private key is only readable
+by root. If your service is running as another user, you can configure
+SIA agent to automatically change the ownership of the private key file
+to the configured user.
+For example, if you want the private key to be owned by `ec2-user`, then
+your `sia_config` configuration file would be as follows:
+
+```
+{
+  "version": "1.0.0",
+  "service": "api",
+  "accounts": [
+    {
+      "domain": "sports",
+      "account": "123456789",
+      "user": "ec2-user"
+    }
+  ]
+}
+```
+
+### SSH Certificate Support
+
+As part of the bootstrap process, SIA also retrieves a host ssh certificate
```

Comment:
I decided to leave this in but I'm not sure about how bastion gets the ssh key from the host as the bootstrapping process isn't open source.